pulsar-common/src/main/java/org/apache/pulsar/common/stats/Rate.java

```diff
@@ -28,6 +28,7 @@ public class Rate {
     private final LongAdder valueAdder = new LongAdder();
     private final LongAdder countAdder = new LongAdder();
     private final LongAdder totalCountAdder = new LongAdder();
+    private final LongAdder totalValueAdder = new LongAdder();
```

Comment:
I wonder if we need to extend this Rate class to add this var, if adding this var significantly impacts memory usage ,especially for clusters with a larger number of metrics objects.

Reply:
My thinking is that it can be reused in other places where OpenTelemetry metrics are being exposed, so it's not useless. When we remove the Prometheus metrics, we will be able to clear out the other two `LongAdder`s instead.

As for memory consumption, it depends on the cardinality of the object itself. I expect consumers, producers and topics to occupy the lions share of the memory here. Since the numbers themselves have to be exposed one way or another, that loss cannot be avoided. I expect many of these `Rate`s will be exposed under OpenTelemetry, and there will be few instances where the new variable is not being read, and thus redundant.

If we need to improve this still, perhaps we can keep the totals only, while replacing the 'relative' adders with simple atomics. Something along the lines of:

```
private final LongAdder totalValueAdder = new LongAdder();
private final AtomicLong totalValueCache = new AtomicLong(); // or an AtomicRef updater
private double valueRate;

public void calculateRate(double period) {
  int current = totalValueAdder.sum();
  int delta = current - totalValueCache.getAndSet(current);
  valueRate = delta / period;
}
```